parquet/src/main/java/org/apache/iceberg/parquet/ReusableArrayData.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.parquet;
+
+public interface ReusableArrayData {
```

Comment:
Was it necessary to make this an `interface`? Because this does allocation, it seems much harder to make it an interface than to make it an abstract class that handles allocation and the values array internally.

Reply:
Yeah, the problem is Spark's base class is an abstract class and Flink's is an interface :/

Reply:
I was thinking about this today. Originally I had assumed that I could not have the flink one inherit from Spark because flink & spark wouldn't want to take dependencies on each other. But since the flink one is an interface, I'll give it a shot and see if I can make this an abstract class.

Reply:
Circled back, and yeah I think this does have to be an interface :/

Reply:
Yes, I think so as well. That just means I'll need to be a bit more careful with the review since it is more complicated than I thought. I should have time today, hopefully.

Reply:
Is it possible that design the ReusableArrayData as a generic class `ReusableArrayData<T>` ? 

and the`FlinkReusableArrayData` and `SparkReusableArrayData` would have a private member which is `ReusableArrayData<Object>` , then both of them could delegate  methods such as capacity(), setNumElements(), getNumElements(). 

Making the ReusableArrayData an interface just confused me a bit, for me it should have a complete implementation , although its type is generic.

Reply:
I don't think so since we can't do multiple inheritence with the class.